### PR TITLE
infrastructure for c++ unit testing

### PR DESCRIPTION
**Pull request by catoverdrive**
@patrick-schultz I've assigned this to you because I wanted your feedback, but I'm happy to spin the wheel if you'd prefer.

cc @cseed

### CODE REVIEW

**Comment by catoverdrive** [comment]
cc @danking can you take a look and make sure I didn't mess up the test reporting?

File: Dockerfile.pr-builder [hunk 1]
```diff
@@ -73,6 +73,8 @@ RUN wget -O qctool_v2.0.1-CentOS6.8-x86_64.tgz http://www.well.ox.ac.uk/~gav/res
     rm -rf qctool_v2.0.1-CentOS6.8-x86_64.tgz && \
     ln -s /qctool/qctool /bin/qctool
 
+RUN wget -O /usr/local/include/catch.hpp https://raw.githubusercontent.com/catchorg/Catch2/master/single_include/catch2/catch.hpp
```
Comment (cseed) [inline Dockerfile.pr-builder]: I think we should pull this from a location with a version.  We can grab from the releases:

https://github.com/catchorg/Catch2/releases

or the tagged versions:

https://github.com/catchorg/Catch2/tree/v2.4.2

Comment (catoverdrive) [reply Dockerfile.pr-builder]: ok, done. both of those link to the same header file.

**Comment by catoverdrive** [comment]
@patrick-schultz I made sure that a failing test would show up correctly on the report, but I forgot to save the link to the build report so I can't show it to you :(

**Comment by patrick-schultz** [comment]
I trust you!

**Pull request merged**